sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -1512,6 +1512,16 @@ object SQLConf {
         "When this conf is not set, the value from `spark.redaction.string.regex` is used.")
       .fallbackConf(org.apache.spark.internal.config.STRING_REDACTION_PATTERN)
 
+  val SQL_EXPLAIN_LEGACY_FORMAT =
```

Comment:
I'd like to do it in followup. We need more discussions about how to explain a DataFrame: do we need new APIs, or more configs? I don't want to block this PR because of it.

Reply:
@cloud-fan Sounds good. Will remove the property.